### PR TITLE
[Monitor Exporter] Replace instrumentationKey with connectionString in builder

### PR DESCRIPTION
**Pull request by srnagar**
This PR replaces the `instrumentationKey()` builder method with `connectionString()` as that's the preferred input to create the exporter instance.
This PR also hides the `endpoint` method on the builder and instead uses the connection string to set the endpoint.

Fixes https://github.com/Azure/azure-sdk-for-java/issues/16705

### CODE REVIEW

File: sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java [hunk 1]
```diff
@@ -134,15 +136,31 @@ public AzureMonitorExporterBuilder configuration(Configuration configuration) {
...
+     * The connection string to use for exporting telemetry events to Azure Monitor.
+     * @param connectionString The connection string for the Azure Monitor resource.
      * @return The updated {@link AzureMonitorExporterBuilder} object.
+     * @throws NullPointerException If the connection string is {@code null}.
+     * @throws IllegalArgumentException If the connection string is invalid.
      */
-    public AzureMonitorExporterBuilder instrumentationKey(String instrumentationKey) {
-        this.instrumentationKey = instrumentationKey;
+    public AzureMonitorExporterBuilder connectionString(String connectionString) {
+        this.instrumentationKey = extractInstrumentationKey(connectionString);
```
Comment (trask) [inline sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java:146]: I think ideally we would also extract `IngestionEndpoint` from the connection string and not expose endpoint() directly

File: sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java [hunk 2]
```diff
@@ -143,22 +143,23 @@ public AzureMonitorExporterBuilder configuration(Configuration configuration) {
      * @throws IllegalArgumentException If the connection string is invalid.
      */
     public AzureMonitorExporterBuilder connectionString(String connectionString) {
-        this.instrumentationKey = extractInstrumentationKey(connectionString);
+        this.instrumentationKey = extractValueFromConnectionString(connectionString, "InstrumentationKey");
```
Comment (trask) [inline sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java:146]: maybe extract once into Map<String, String> once and pull values from there? not a big deal tho

File: sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java [hunk 3]
```diff
@@ -134,15 +136,32 @@ public AzureMonitorExporterBuilder configuration(Configuration configuration) {
...
      * @return The updated {@link AzureMonitorExporterBuilder} object.
+     * @throws NullPointerException If the connection string is {@code null}.
+     * @throws IllegalArgumentException If the connection string is invalid.
      */
-    public AzureMonitorExporterBuilder instrumentationKey(String instrumentationKey) {
-        this.instrumentationKey = instrumentationKey;
+    public AzureMonitorExporterBuilder connectionString(String connectionString) {
+        this.instrumentationKey = extractValueFromConnectionString(connectionString, "InstrumentationKey");
+        this.endpoint(extractValueFromConnectionString(connectionString, "IngestionEndpoint"));
         return this;
```
Comment (trask) [inline sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java:148]: if the `IngestionEndpoint` key is absent, ideally should default to `https://dc.services.visualstudio.com`

File: sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java (hunk 2 above)
Comment (srnagar) [reply sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java]: Done

File: sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java (hunk 3 above)
Comment (srnagar) [reply sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java:156]: Updated to use the default endpoint if connection string doesn't have one. Although, I am curious when would a connectionString not have the IngestionEndpoint. If it doesn't have one, will that qualify as a valid connectionString?

Comment (trask) [reply sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/src/main/java/com/microsoft/opentelemetry/exporter/azuremonitor/AzureMonitorExporterBuilder.java:156]: that's a good question, when we originally implemented connection strings, the Portal U/X only included IngestionEndpoint for gov clouds, so the default was important, but now the Portal U/X always includes IngestionEndpoint, so the default is likely not needed anymore. I'll start an internal thread and add you to see if we can find out.

**Review by trask — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Review by joshfree — APPROVED** [review]

**Pull request merged**